core/common/src/MapLayerSettings.ts

```diff
@@ -349,24 +403,93 @@ export class MapLayerSettings {
+    super(name, visible, transparency, transparentBackground);
+    this.modelId = modelId;
+  }
+  /** Construct from JSON, performing validation and applying default values for undefined fields. */
+  public static override fromJSON(json: ModelMapLayerProps): ModelMapLayerSettings {
+    const transparentBackground = (json.transparentBackground === undefined) ? true : json.transparentBackground;
+    return new this(json.modelId, json.name, json.visible, json.transparency, transparentBackground);
+  }
+  /** return JSON representation of this MapLayerSettings object */
+  public override toJSON(): ModelMapLayerProps {
+    const props = super.toJSON();
+    return { ... props, modelId: this.modelId };
+  }
+  /** Create a copy of this MapLayerSettings, optionally modifying some of its properties.
+   * @param changedProps JSON representation of the properties to change.
+   * @returns A MapLayerSettings with all of its properties set to match those of `this`, except those explicitly defined in `changedProps`.
+   */
+  public clone(changedProps: Partial<ModelMapLayerProps>): ModelMapLayerSettings {
+    return ModelMapLayerSettings.fromJSON(this.cloneProps(changedProps));
+  }
+  /** @internal */
+  protected override cloneProps(changedProps: Partial<ModelMapLayerProps>): ModelMapLayerProps {
+    const baseProps = super.cloneProps(changedProps);
+    return {... baseProps, modelId: undefined !== changedProps.modelId ? changedProps.modelId : this.modelId };
+  }
+  /** @internal */
+  public override displayMatches(other: MapLayerSettings): boolean {
+    if (!(other instanceof ModelMapLayerSettings) || !super.displayMatches(other))
+      return false;
+
+    return this.modelId === other.modelId;
+  }
+  /** Return true if all sublayers are invisible (always false as model layers do not include sublayers). */
+  public get allSubLayersInvisible(): boolean {
+    return false;
+  }
 }
 
 /** JSON representation of a [[BaseMapLayerSettings]].
  * @beta
  */
-export interface BaseMapLayerProps extends MapLayerProps {
+export interface BaseMapLayerProps extends ImageMapLayerProps {
   provider?: BackgroundMapProviderProps;
 }
 
-/** A [[MapLayerSettings]] that can serve as the base layer for a [[MapImagerySettings]].
+/** A [[ImageMapLayerSettings]] that can serve as the base layer for a [[MapImagerySettings]].
  * The base layer supports all of the same options as any other layer, but also allows for simplified configuration based
  * on a small set of known supported [[BackgroundMapProvider]]s like [Bing Maps](https://www.microsoft.com/en-us/maps).
  * If the base layer was configured from such a provider, that information will be preserved and can be queried; this allows
  * the imagery provider and/or type to be easily modified.
  * @see [[MapImagerySettings.backgroundBase]].
  * @beta
  */
-export class BaseMapLayerSettings extends MapLayerSettings {
+export class BaseMapLayerSettings extends ImageMapLayerSettings {
```

Comment:
@mdastous-bentley shouldn't BaseMapLayerSettings.cloneProps take a Partial<BaseMapLayerProps> so that the caller can change the provider?

Reply:
Made this change -- resolve unless you object.